### PR TITLE
Add PHPStan

### PR DESCRIPTION
**Pull request by divine**
Add PHPStan

Closes: https://github.com/jenssegers/laravel-mongodb/issues/1930

Fixes will be added with another PR to keep everything simple as possible.

### CODE REVIEW

**Review by jenssegers — APPROVED** [review]

File: .github/workflows/build-ci.yml [hunk 1]
```diff
@@ -27,7 +27,27 @@ jobs:
...
+                uses: shivammathur/setup-php@v2
+                with:
+                    php-version: ${{ matrix.php }}
+                    extensions: curl,mbstring
+                    tools: composer
+                    coverage: none
+                    ini-values: memory_limit=-1
+            -   name: Install dependencies
+                run: composer install --no-interaction
+            -   name: Run PHPStan Analysis
```
Comment (Smolevich) [inline .github/workflows/build-ci.yml:49]: Maybe to mark as not required step

**Comment by Treggats** [comment]
@divine would you be ok with it if I continue'd your work with this PR?

**Comment by divine** [comment]
> @divine would you be ok with it if I continue'd your work with this PR?

Hello,

Of course you can continue to work on this. Honestly, it's quite old PR, at that time there were like 300 errors that needed to be fixed, not sure how much is work currently is needed.

Thanks!

**Comment by Treggats** [comment]
> > @divine would you be ok with it if I continue'd your work with this PR?
> 
> Hello,
> 
> Of course you can continue to work on this. Honestly, it's quite old PR, at that time there were like 300 errors that needed to be fixed, not sure how much is work currently is needed.
> 
> Thanks!

The amount of work depends on the level. On level 5 there are around 64 issues and on level 9, which is the max, there are 384 errors.
But for now I've set it to level 2, with a baseline. So it would ignore certain errors. Makes it easier to ease in to fixing those.

**Comment by GromNaN** [comment]
Closing in favor of #2664

**Pull request closed without merging**